analysis/import/TokeiImporter/src/test/resources/tokei_without_inner.json

```diff
@@ -0,0 +1,20 @@
+{
+  "Markdown": {
+    "blanks": 0,
+    "code": 1965,
+    "comments": 0,
+    "reports": [
+      {
+        "stats": {
+          "blanks": 0,
+          "code": 450,
+          "comments": 0,
+          "blobs": {}
+        },
+        "name": "foo/bar/CHANGELOG.md"
+      }
+    ],
+    "children": {},
```

Comment:
example for children
```
"children": {
            "Bash": [
                {
                    "stats": { "blanks": 0, "code": 2, "comments": 3, "blobs": {} },
                    "name": "./README.md"
                },
                {
                    "stats": { "blanks": 6, "code": 20, "comments": 25, "blobs": {} },
                    "name": "./gh-pages/_docs/01-03-installation.md"
                },
                {
                    "stats": { "blanks": 0, "code": 6, "comments": 6, "blobs": {} },
                    "name": "./gh-pages/_docs/04-04-scmlogparser.md"
                },
                {
                    "stats": { "blanks": 0, "code": 19, "comments": 16, "blobs": {} },
                    "name": "./gh-pages/_docs/01-01-quick-start-guide.md"
                },
                {
                    "stats": { "blanks": 0, "code": 1, "comments": 0, "blobs": {} },
                    "name": "./gh-pages/_docs/04-09-jasomeimporter.md"
                },
                {
                    "stats": { "blanks": 0, "code": 5, "comments": 0, "blobs": {} },
                    "name": "./gh-pages/_docs/04-08-codemaatimporter.md"
                },
                {
                    "stats": { "blanks": 0, "code": 1, "comments": 0, "blobs": {} },
                    "name": "./gh-pages/_docs/04-06-tokeiImporter.md"
                },
                {
                    "stats": { "blanks": 5, "code": 8, "comments": 6, "blobs": {} },
                    "name": "./gh-pages/_posts/how-to/2019-10-23-analyze_java_project_and_git_log.md"
                }
            ],
```

Reply:
I personally would also check for children and write tests for both: one without children, one with children.

Reply:
But what are the children used for? How can a file type like `Bash` have children?

Reply:
Valid point. I guess we could use a completely separate folder that is used instead?

Reply:
tokei.json contains file-endings (like Bash, Java, Markdown). Each contains an array of objects, containing the path of the file and it's metrics in `stats` You also have the aggregated metrics in the file-endings object, but we aggregate those ourselves in the visualization. So basically we only need stats.
It seems like that you can count Javascript-Code in markdown files etc, but I don't think we used those in the previous version of tokei.

It makes sense to just generate two jsons of the same folder and compare the cc.jsons.